src/Controller/NotificationDefinitionConfirmationController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\Entity\NotificationDefinition;
+use App\Response\ApiResponse;
+use App\Response\NotFoundApiResponse;
+use App\Response\SuccessApiResponse;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Routing\Annotation\Route;
+
+class NotificationDefinitionConfirmationController extends AbstractController
+{
+    /**
+     * Perform a findOneBy() where the token property matches {token}.
+     */
+    #[Route("/api/confirm/{token}", name: 'token-confirmation', methods: 'GET')]
+    public function __invoke(
+        EntityManagerInterface $entityManager,
+        string $token
+    ): ApiResponse {
+
+        $repository = $entityManager->getRepository(NotificationDefinition::class);
+        $entity = $repository->findOneBy(['confirmationToken' => $token]);
```

Comment:
W repository bym zrobił metodę findOneByConfirmationToken i w niej wkleił to co masz tu, a tutaj wywołał tą metodę